download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
The question is more, what's the meaning of "adding to index" ? Does this mean that the file pointed by this path file is copied or referenced? If the former, then I'd suggest to simply put an InputStream, if the later then a file. And the documentation should explain whether this index copies or references the content.